docs/core-concepts/core-sdk/cli/dotnet-test-protocol.md

```diff
@@ -0,0 +1,45 @@
+dotnet-test communication protocol
+===================================
+
+## Introduction
+Anytime you pass a port to dotnet test, we will run in design time. That means that dotnet test will connect to that port
+using TCP and will then exchange an established set of messages with whatever else is connected to that port. When this happens, the runner
+also receives a new port that dotnet test will use to communicate with it. The reason why the runner also uses TCP to
+communicate with dotnet test is because in design mode, it is not sufficient to just output results to the console. The 
+command needs to send the adapter structure messages containing the results of the test execution.
+
+### Communication protocol in design time.
+
+1. Because during design time, dotnet test connects to a port when it starts up, the adapter needs to be listening on 
```

Comment:
nit: remove comma after design time and add semi-colon before otherwise
Do you have to explain yourself here? Otherwise, I'd rewrite the sentence that starts with "We did it like this" to just say how it behaves.